File: features/lender/src/main/java/com/nivasafinance/features/lender/lenderoffice/exception/LenderOfficeExceptionFactory.java
package com.nivasafinance.features.lender.lenderoffice.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

import java.io.Serial;
import java.util.UUID;

public class LenderOfficeExceptionFactory {

    public static LenderOfficeOperationException createFailed(MessageSource messageSource) {
        return new LenderOfficeOperationException("error.lender.office.operation.failed", messageSource);
    }

    public static LenderOfficeNotFoundException lenderOfficeNotFound(UUID id, MessageSource messageSource) {
        String message = ExceptionUtils.createLocalizedMessage(
                "error.lender.office.not.found",
                new Object[]{id.toString()},
                messageSource
        );
        return new LenderOfficeNotFoundException(message);
    }

    public static LenderOfficeNotFoundException lenderOfficeKeyNotFound(String key, MessageSource messageSource) {
        String message = ExceptionUtils.createLocalizedMessage(
                "error.lender.office.key.not.found",
                new Object[]{key},
                messageSource
        );
        return new LenderOfficeNotFoundException(message);
    }
}

File: features/lender/src/main/java/com/nivasafinance/features/lender/lenderoffice/exception/LenderOfficeExceptionFactory.kt
package com.nivasafinance.features.lender.lenderoffice.exception

import exception.ExceptionUtils
import org.springframework.context.MessageSource
import java.util.UUID

object LenderOfficeExceptionFactory {

    fun createFailed(messageSource: MessageSource): LenderOfficeOperationException {
        return LenderOfficeOperationException("error.lender.office.operation.failed", messageSource)
    }

    fun lenderOfficeNotFound(id: UUID, messageSource: MessageSource): LenderOfficeNotFoundException {
        val message = ExceptionUtils.createLocalizedMessage(
            "error.lender.office.not.found",
            arrayOf(id.toString()),
            messageSource
        )
        return LenderOfficeNotFoundException(message)
    }

    fun lenderOfficeKeyNotFound(key: String, messageSource: MessageSource): LenderOfficeNotFoundException {
        val message = ExceptionUtils.createLocalizedMessage(
            "error.lender.office.key.not.found",
            arrayOf(key),
            messageSource
        )
        return LenderOfficeNotFoundException(message)
    }
}

class LenderOfficeOperationException(
    messageKey: String,
    messageSource: MessageSource
) : RuntimeException(
    ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource)
)
